app/models/target.rb

```diff
@@ -3,4 +3,11 @@ class Target < ApplicationRecord
   belongs_to :user
 
   validates :title, :radius, :latitude, :longitude, :topic_id, presence: true
+  
+  validate :user_max_targets
+
+  def user_max_targets
+    user = User.find(self.user_id)
```

Comment:
You can remove this line. As this user is the user it self
Try this:
`errors.add(:targets, "You are allowed to create only 3 targets") if targets.length >= 3`